sheets/participationsheet.py

```diff
@@ -0,0 +1,333 @@
+from typing import List, Any
+
+from sheets.commandoutput import GivePointsOutput
+import sheets.base26 as base26
+from sheets.semester import Semester
+
+sheets = None
+SHEETID = None
+
+"""
+    Gets the column name of the column to the right of the column provided.
+    If the last column ends with 'Z', returns AA, returns AB, ... 'AZ' 
+"""
+
+
+def next_column_name(col: str) -> str:
+    return base26.str_from_b26(base26.tob26(col) + 1)
+
+
+class Subsheet:
+    EVENTSROW = 1
+    TOTAL = 2
+    MEAN = 3
+    MEDIAN = 4
+    QUARTILE1 = 5
+    # Represents what row user data is stored at on the sheet. This is after Name, Total, Mean, Median, 1st Quartile
+    USERDATASTART = 6
+    # Where names show up
+    NAME = "A"
+    CODENAME = "B"
+    EVENTSSTART = "F"
+    GID = "0"
+    USERSTATSMATH: list[str] = list()
+
+    sheetname = ""
+
+    def __init__(self, name: str):
+        self.sheetname = name
+
+    # Allow loading from JSON.
+    def apply(self, jsonmap: dict):
+        self.EVENTSROW = jsonmap["EVENTSROW"]
+        self.TOTAL = jsonmap["TOTAL"]
+        self.MEAN = jsonmap["MEAN"]
+        self.MEDIAN = jsonmap["MEDIAN"]
+        self.QUARTILE1 = jsonmap["QUARTILE1"]
+        self.USERDATASTART = jsonmap["USERDATASTART"]
+        self.NAME = jsonmap["NAME"]
+        self.EVENTSSTART = jsonmap["EVENTSSTART"]
+        self.CODENAME = jsonmap["CODENAME"]
+        self.GID = jsonmap["GID"]
+        self.USERSTATSMATH = jsonmap["USERSTATSMATH"]
+
+    """
+        Gets the next empty column in the desired sheet. A column is empty if the first row does not contain values.
+        Inputting a None or empty string will automatically check column 'A' first.
+        Providing a column such as 'A' will check the next column ('B' in this case)
+    """
+
+    def next_open_column(self, col: str) -> str:
+        col = next_column_name(col)
+        result = sheets.sheet.values().get(spreadsheetId=SHEETID, range=self.named_range() + col + "1").execute()
+        # Result will not contain a values key if the column is open.
+        if "values" not in result:
+            return col
+        # If not open, check the next column.
+        else:
+            return self.next_open_column(col)
+    """
+        Gets the next empty user name row.
+    """
+    def next_open_user_row(self):
+        return self.USERDATASTART + len(self.get_names())
+    """
```

Comment:
Move this docstring so it touches the method it documents (right now there is a blank line)